sim/src/proto/valueType.cpp

```diff
@@ -0,0 +1,124 @@
+#include <proto/valueType.h>
+
+namespace proto {
+
+std::string valStoredTypeName(const bento::protos::Value_Primitive& val) {
```

Comment:
Might be possible to deduce the value from the `Value_Primitive`'s descriptor:
https://github.com/protocolbuffers/protobuf/issues/4215#issuecomment-359991115